odl/solvers/advanced/proximal_operators.py

```diff
@@ -124,14 +133,180 @@ def cconj_prox_factory(step_size):
         Returns
         -------
         proximal : `Operator`
-            The proximal operator of ``prox[a * f^*](x)``
+            The proximal operator of ``s * F^*`` where ``s`` is the step size
         """
-        prox_other = step_size * proximal(1.0 / step_size) * (1.0 / step_size)
+        prox_other = (step_size * prox_factory(1.0 / step_size) *
+                      (1.0 / step_size))
         return IdentityOperator(prox_other.domain) - prox_other
 
     return cconj_prox_factory
 
 
+def proximal_translation(prox_factory, y):
+    """Calculate the proximal of the translated function F(x - y).
+
+    This is calculated according to the rule
+
+        prox[s * F( . - y)](x) = y + prox[s * F](x - y)
+
+    where ``a`` is the translation, and ``s`` is the step size.
+
+    Parameters
+    ----------
+    prox_factory : `callable`
+        A factory function that, when called with a step size, returns the
+        proximal operator of ``F``
+    y : Element in domain of F
+
+    Returns
+    -------
+    prox : `callable`
+        Factory for the proximal operator to be initialized
+
+    Notes
+    -----
+    For reference on the identity used, see [ComPes2011]_.
+    """
+
+    def translation_prox_factory(step_size):
+        """Create proximal for the translation with a given step_size.
+
+        Parameters
+        ----------
+        step_size : positive `float`
+            Step size parameter
+
+        Returns
+        -------
+        proximal : `Operator`
+            The proximal operator of ``s * F( . - y)`` where ``s`` is the
+            step size
+        """
+
+        return (ConstantOperator(y) + prox_factory(step_size) *
+                (IdentityOperator(y.space) - ConstantOperator(y)))
+
+    return translation_prox_factory
+
+
+def proximal_arg_scaling(prox_factory, a):
+    """Calculate the proximal of function F(x * 1/a).
+
+    This is calculated according to the rule
+
+        prox[s * F( . * 1/a)](x) = a * prox[s/a^2 * F ](x * 1/a)
+
+    where ``a`` scaling parameter, and ``s`` is the step size.
+
+    Parameters
+    ----------
+    prox_factory : `callable`
+        A factory function that, when called with a step size, returns the
+        proximal operator of ``F``
+    a : non-zero `float`
+        Scaling parameter
+
+    Returns
+    -------
+    prox : `callable`
+        Factory for the proximal operator to be initialized
+
+    Notes
+    -----
+    For reference on the identity used, see [ComPes2011]_.
+    """
+
```

Comment:
The input parameter the we take is ```y``` (it was a typo in the Doc where it says ```a```). Why is a translation, so it lives in the domain of the function. The "problem" is that the proximal factory cannot tell me what that domain is, so if you want to check that you have to create a proximal operator and then ask that one for the domain. Should this be done?

The only float value that we take in this class is the step length. Should I add this to all proximal factories? (basically all of them take take a step length as input)